session/session_test.go

```diff
@@ -48,11 +52,60 @@ func TestReceiveEventWhenSubscribing(t *testing.T) {
 		if event != SessionDisconnected {
 			t.Error("Expected to receive disconnected: ", event)
 		}
+	case <-time.After(5 * time.Second):
+		t.Error("Failed to receive event")
+	}
 
-		if <-events != SessionReconnected {
+	select {
+	case event := <-events:
+		if event != SessionReconnected {
 			t.Error("Expected to receive reconnected: ", event)
 		}
-	case <-time.After(100 * time.Millisecond):
+	case <-time.After(5 * time.Second):
 		t.Error("Failed to receive event")
 	}
 }
+
+func TestResumeZKSessionWithValidSession(t *testing.T) {
+	proxy := CreateProxy(t)
+	defer proxy.Delete()
+
+	store, err := NewZKSession("localhost:27445", 200*time.Millisecond, nil)
+	if err != nil {
+		t.Error("Failed to connect to Zookeeper: ", err)
+	}
+
+	existingClientId, err := store.ClientId().Save()
+	if err != nil {
+		t.Error("Failed to save clientId: ", err)
+	}
+
+	store.Delete("localhost:27445", -1)
```

Comment:
Ive tried calling `NewZKSession()` after `store.Close()` and it cannot reconnect to zookeeper either. 

With `store.Delete()` ive tried using the path `/github.com/Shopify/gozk-recipes/session` found from printing `zookeeper.Event.Path` in session.go but it doesnt seem to work either. `session_test.go:85: Failed to delete existing session:  zookeeper: delete "/github.com/Shopify/gozk-recipes/session": no node`